libcaf_core/caf/policy/work_stealing.hpp

```diff
@@ -85,7 +85,8 @@ class work_stealing : public unprofiled {
       return nullptr;
     }
     // roll the dice to pick a victim other than ourselves
-    size_t victim = d(self).rengine() % (p->num_workers() - 1);
+    std::uniform_int_distribution<size_t> uniform(0, p->num_workers() - 2);
```

Comment:
This is has nothing to do with your patch (which looks valid to me), but I wonder why we are not drawing a value from *[0, N)* where *N* is the number of workers. Why are we using *N-1*? There is a check in place afterwards where we take an arbitrary worker in case we picked ourself.

Not sure if I am missing something obvious here.